Add bandwidth to NodeResources

freva opened the pull request:
`Flavor` now only has `cost` and `type`. Currently we don't really need type since we only use `DOCKER_CONTAINER` and `BARE_METAL` (or `VIRTUAL_MACHINE` in aws), but that is is 1-to-1 with `NodeType::isDockerHost`. Though type could be added to `NodeResources`, as it may make sense in the future to ask to be allocated on a `DOCKER_CONTAINER` or something else :man_shrugging: 

As for cost, I think it should be calculated by some component that can be injected into node-repo?

freva committed:
Add bandwidth to NodeResources

freva committed:
Add toSlime

bratseth reviewed (changes requested):
I think we should use Gbps and use 10Gbps as the default (otherwise 0 has some special meaning and we'd need to document that). Otherwise LGTM.

freva committed:
bandwidthMbps -> bandwidthGbps

freva committed:
Deprecate NodeResources constructors without bandwidth

freva committed:
Stop using deprecated NodeResources constructors

freva committed:
Merge branch 'master' into freva/add-bandwidth-to-node-resources

# Conflicts:
#	node-repository/src/test/java/com/yahoo/vespa/hosted/provision/maintenance/LoadBalancerExpirerTest.java

freva committed:
Update default values when deserializing resources

freva commented:
@bratseth PTAL.

This PR will enforce bandwidth resources in allocation (actual enforcement on host is for later). Therefore, I changed the default value to 300 Mbps (largest hosts have 32 nodes, so 10 Gbps / 32 = 312.5 Mbps). I also changed the method to use Gbps, but I the flavor config and `/nodes/v2` API already used Mbps, so there is some conversion between those and `NodeResources`.

This must be merged with the corresponding internal PR.

bratseth reviewed (approved):
Great!
